hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetworkTopologyImpl.java

```diff
@@ -754,7 +754,7 @@ public List<? extends Node> sortByDistanceCost(Node reader,
       List<? extends Node> nodes, int activeLen) {
     /** Sort weights for the nodes array */
     if (reader == null) {
-      return nodes;
+      return Collections.emptyList();
```

Comment:
Javadoc needs updating to reflect the behavior.

Reply:
Updated, thanks!

Reply:
Note that OM will not make `sortDatanodes` request to SCM if client is unknown:

https://github.com/apache/ozone/blob/72ab6a7a53c2603bab27035dfac510d469d2f402/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java#L1815-L1816

So this part of the change may not be necessary.